packages/@react-spectrum/table/test/Table.test.js

```diff
@@ -1556,4 +1568,357 @@ describe('Table', function () {
+        expect(rows).toHaveLength(3);
+
+        expect(rows[0].style.top).toBe('0px');
+        expect(rows[0].style.height).toBe('34px');
+        expect(rows[1].style.top).toBe('0px');
+        expect(rows[1].style.height).toBe('49px');
+        expect(rows[2].style.top).toBe('49px');
+        expect(rows[2].style.height).toBe('49px');
+
+        for (let cell of [...rows[1].childNodes, ...rows[2].childNodes]) {
+          expect(cell.style.top).toBe('0px');
+          expect(cell.style.height).toBe('48px');
+        }
+      });
+
+      it('should layout rows with default height in large scale', function () {
+        let tree = renderTable({}, 'large');
+        let rows = tree.getAllByRole('row');
+        expect(rows).toHaveLength(3);
+
+        expect(rows[0].style.top).toBe('0px');
+        expect(rows[0].style.height).toBe('40px');
+        expect(rows[1].style.top).toBe('0px');
+        expect(rows[1].style.height).toBe('65px');
+        expect(rows[2].style.top).toBe('65px');
+        expect(rows[2].style.height).toBe('65px');
+
+        for (let cell of [...rows[1].childNodes, ...rows[2].childNodes]) {
+          expect(cell.style.top).toBe('0px');
+          expect(cell.style.height).toBe('64px');
+        }
+      });
+
+      it('should layout rows with a custom rowHeight', function () {
+        let tree = renderTable({rowHeight: 72});
+        let rows = tree.getAllByRole('row');
+        expect(rows).toHaveLength(3);
+
+        expect(rows[0].style.top).toBe('0px');
+        expect(rows[0].style.height).toBe('34px');
+        expect(rows[1].style.top).toBe('0px');
+        expect(rows[1].style.height).toBe('73px');
+        expect(rows[2].style.top).toBe('73px');
+        expect(rows[2].style.height).toBe('73px');
+
+        for (let cell of [...rows[1].childNodes, ...rows[2].childNodes]) {
+          expect(cell.style.top).toBe('0px');
+          expect(cell.style.height).toBe('72px');
+        }
+      });
+
+      it('should support variable row heights with rowHeight="auto"', function () {
+        let scrollHeight = jest.spyOn(window.HTMLElement.prototype, 'scrollHeight', 'get')
+          .mockImplementation(function () {
+            return this.textContent === 'Foo 1' ? 64 : 48;
+          });
+        
+        let tree = renderTable({rowHeight: 'auto'});
+        let rows = tree.getAllByRole('row');
+        expect(rows).toHaveLength(3);
+
+        expect(rows[1].style.top).toBe('0px');
+        expect(rows[1].style.height).toBe('65px');
+        expect(rows[2].style.top).toBe('65px');
+        expect(rows[2].style.height).toBe('49px');
+
+        for (let cell of rows[1].childNodes) {
+          expect(cell.style.top).toBe('0px');
+          expect(cell.style.height).toBe('64px');
+        }
+
+        for (let cell of rows[2].childNodes) {
+          expect(cell.style.top).toBe('0px');
+          expect(cell.style.height).toBe('48px');
+        }
+
+        scrollHeight.mockRestore();
```

Comment:
shouldn't need to call this, were you running into an issue if you didn't?
otherwise you can make an afterEach https://jestjs.io/docs/en/jest-object#jestrestoreallmocks

Reply:
if I didn't restore it, wouldn't it leak into other tests?

Reply:
yeah, just noticed it was a spyOn a global, but this should really be cleaned up in an after hook, otherwise we run the risk that should this test fail, it will leak into following tests and cascade the failure